In [3]:
# import necessary library
import pandas as pd # Pandas for data manipulation and analysis using tables (DataFrames)
import numpy as np # Numpy for numerical computing and efficient array operations.
import matplotlib.pyplot as plt #matplotlib.pyplot: Used to create a wide range of basic visualizations
import seaborn as sns # seaborn: Advance visualizations for statistics.
import warnings as wr
wr.filterwarnings("ignore") # warnings: To handle unnecessary warnings

In [4]:
# Load the dataset
df = pd.read_csv(r"C:\Users\Din islam\Downloads\diabetics-prediction-\diabetes.csv")

In [ ]:
# Here, my dataset is in csv format. So, I used read_csv() function to  load the dataset

In [6]:
# To see the first 5 row from the dataset
df.head(5)

Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
0            6      148             72             35        0  33.6   
1            1       85             66             29        0  26.6   
2            8      183             64              0        0  23.3   
3            1       89             66             23       94  28.1   
4            0      137             40             35      168  43.1   

   DiabetesPedigreeFunction  Age  Outcome  
0                     0.627   50        1  
1                     0.351   31        0  
2                     0.672   32        1  
3                     0.167   21        0  
4                     2.288   33        1

In [5]:
# Checking the shape of the dataset to see how many rows and columns in the dataset
df.shape

(768, 9)

#### In this, dataset there are 768 rows and 9 columns

# - Data Preprocessing & Cleaning


In [7]:
### To know the overall information about the dataset
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 768 entries, 0 to 767
Data columns (total 9 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Pregnancies               768 non-null    int64  
 1   Glucose                   768 non-null    int64  
 2   BloodPressure             768 non-null    int64  
 3   SkinThickness             768 non-null    int64  
 4   Insulin                   768 non-null    int64  
 5   BMI                       768 non-null    float64
 6   DiabetesPedigreeFunction  768 non-null    float64
 7   Age                       768 non-null    int64  
 8   Outcome                   768 non-null    int64  
dtypes: float64(2), int64(7)
memory usage: 54.1 KB


### Here, in the dataset , there is no missing value at any feature and not categorical feature.

In [8]:
#Checking null value Present or not by using pandas
df.isnull().sum()

Pregnancies                 0
Glucose                     0
BloodPressure               0
SkinThickness               0
Insulin                     0
BMI                         0
DiabetesPedigreeFunction    0
Age                         0
Outcome                     0
dtype: int64

### df.isnull().sum() shows how many null value present in each feature